ZuluContent/Engines/Harvest/Core/HarvestSystem.cs

```diff
@@ -1,504 +1,502 @@
-		{
-			HarvestDefinition def = null;
-
-			for ( int i = 0; def == null && i < m_Definitions.Count; ++i )
-			{
-				HarvestDefinition check = m_Definitions[i];
-
-				if ( check.Validate( tileID ) )
-					def = check;
-			}
-
-			return def;
-		}
-
-		public virtual void StartHarvesting( Mobile from, Item tool, object toHarvest )
-		{
-			if ( !CheckHarvest( from, tool ) )
-				return;
-
-			int tileID;
-			Map map;
-			Point3D loc;
-
-			if ( !GetHarvestDetails( from, tool, toHarvest, out tileID, out map, out loc ) )
-			{
-				OnBadHarvestTarget( from, tool, toHarvest );
-				return;
-			}
-
-			HarvestDefinition def = GetDefinition( tileID );
-
-			if ( def == null )
-			{
-				OnBadHarvestTarget( from, tool, toHarvest );
-				return;
-			}
-
-			if ( !CheckRange( from, tool, def, map, loc, false ) )
-				return;
-			else if ( !CheckResources( from, tool, def, map, loc, false ) )
-				return;
-			else if ( !CheckHarvest( from, tool, def, toHarvest ) )
-				return;
-
-			object toLock = GetLock( from, tool, def, toHarvest );
-
-			if ( !from.BeginAction( toLock ) )
-			{
-				OnConcurrentHarvest( from, tool, def, toHarvest );
-				return;
-			}
-
-			new HarvestTimer( from, tool, this, def, toHarvest, toLock ).Start();
-			OnHarvestStarted( from, tool, def, toHarvest );
-		}
-
-		public virtual bool GetHarvestDetails( Mobile from, Item tool, object toHarvest, out int tileID, out Map map, out Point3D loc )
-		{
-			if ( toHarvest is Static && !((Static)toHarvest).Movable )
-			{
-				Static obj = (Static)toHarvest;
-
-				tileID = (obj.ItemID & 0x3FFF) | 0x4000;
-				map = obj.Map;
-				loc = obj.GetWorldLocation();
-			}
-			else if ( toHarvest is StaticTarget )
-			{
-				StaticTarget obj = (StaticTarget)toHarvest;
-
-				tileID = (obj.ItemID & 0x3FFF) | 0x4000;
-				map = from.Map;
-				loc = obj.Location;
-			}
-			else if ( toHarvest is LandTarget )
-			{
-				LandTarget obj = (LandTarget)toHarvest;
-
-				tileID = obj.TileID;
-				map = from.Map;
-				loc = obj.Location;
-			}
-			else
-			{
-				tileID = 0;
-				map = null;
-				loc = Point3D.Zero;
-				return false;
-			}
-
-			return map != null && map != Map.Internal;
-		}
-	}
+    {
+        private List<HarvestDefinition> m_Definitions;
+
+        public List<HarvestDefinition> Definitions
+        {
+            get { return m_Definitions; }
+        }
+
+        public HarvestSystem()
+        {
+            m_Definitions = new List<HarvestDefinition>();
+        }
+
+        public virtual bool CheckTool(Mobile from, Item tool)
+        {
+            bool wornOut = tool == null || tool.Deleted ||
+                           tool is IUsesRemaining && ((IUsesRemaining) tool).UsesRemaining <= 0;
+            bool equipped = tool.Parent == from;
+
+            if (wornOut)
+                from.SendLocalizedMessage(1044038); // You have worn out your tool!
+
+            if (!equipped)
+                from.SendAsciiMessage(33, "You must equip it to mine.");
```

Comment:
The Cliloc for this is `502641` "You must equip this item to use it."

I'm trying to make an effort to use the Clilocs if possible, so if someone wants to use unicode it still works